src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,140 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    static class Node<T>{
+        T item;
+        Node<T> prev;
+        Node<T> next;
+
+        public Node(T item) {
+            this.item = item;
+        }
+    }
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(value);
+        if (size == 0){
+            head = tail = newNode;
+        } else {
+            tail.next = newNode;
+            tail = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new ArrayIndexOutOfBoundsException();
+        }
+        Node<T> newNode = new Node<>(value);
+        if (head == null){
+            head = tail = newNode;
+        } else if (index == 0){
+            newNode.next = head;
+            head = newNode;
+        } else if (index == size){
+            newNode.prev = tail;
+            tail = newNode;
+        } else {
+            Node<T> current = gettNodeByIndex(index);
+            newNode.next = current.next;
+            current.next = newNode;
+        }
+        size++;
+    }
+
+    private Node<T> gettNodeByIndex(int index) {
+        Node<T> current = head;
+        for (int i = 0; i < index; i++) {
+            current = current.next;
+        }
+        return current;
     }
 
     @Override
     public void addAll(List<T> list) {
+        Node<T> current = head;
+        for (int i = 0; i < size; i++) {
+            list.add(current.item);
+            current = current.next;
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index < 0 || index > size){
+            throw new ArrayIndexOutOfBoundsException();
+        }
+        return gettNodeByIndex(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index);
+        Node<T> current = gettNodeByIndex(index);
+        current.item = value;
+        return current.item;
+
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        T object = null;
+        if (index < 0 || index > size) {
+            throw new ArrayIndexOutOfBoundsException();
```

Comment:
In the `remove(int index)` method, the condition should be `index >= size` instead of `index > size` to correctly handle the case where the index is out of bounds.